### ISSUE TITLE
Lagging while typing

### ISSUE THREAD

**Issue by jt196** [laurent22/joplin#3893]
After the upgrade to 1.2, Joplin has a noticeable lag when editing a note. When I'm typing in text, there is a delay between typing, and the letters coming up on the screen. There is also a delay in the rendered markup being displayed but that's not quite as important as the raw text.

This is running on a fast machine, with no issues with text lagging on any other apps.

## Environment

Joplin version: 1.2.6 (happened since 1.2)
Platform: Mac
OS specifics: Catalina

## Steps to reproduce

1. Open note
2. Type text into note
3. Observe delay

## Describe what you expected to happen

Zero delay or unnoticeable

## Logfile

n/a

**Comment by JackGruber** [comment]
On one Windows 10 device i have the same error. 

![ejGwbzDWgb](https://user-images.githubusercontent.com/24863925/96478901-82fcb480-1238-11eb-822a-508aaa3c7f23.gif)
But the console and the log files show no entries at this time.

I have the lag with and without der preview window on Joplin 1.2.6 and 1.3.5.

**Comment by jt196** [comment]
Yep that's the same as mine, exactly this. 

I opened up [a question on the Joplin forum](https://discourse.joplinapp.org/t/delay-lag-while-typing/11789/6) as there was no response here, maybe you want to add this there as well. 

I'd love to get this fixed. Currently editing stuff in the VS Code version as it makes it really hard to type when you're not getting up to date feedback on your writing.

**Comment by jt196** [comment]
```

DevTools failed to parse SourceMap: file:///Applications/Joplin.app/Contents/Resources/app.asar/gui/ResourceScreen.js.map

---

/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:11494 Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://fb.me/react-async-component-lifecycle-hooks for details.

* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://fb.me/react-derived-state
* Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

Please update the following components: Connect(NoteTextViewerComponent), Connect(NoteToolbar), Connect(Toolbar), Connect(ToolbarBaseComponent)
```
In the log file, I don't know if this is normal behaviour but this stuff appears in the debugging mode every 4 seconds or so while I'm editing. (Could it be that it's saving/synching too often?):

```
2020-10-19 09:22:47: "Scheduling sync operation...", "30000"
2020-10-19 09:22:50: "Saving note...", "{"id":"0e6e82daa78241c7bfe71016aa5911ea", <contents of note here>
```

**Comment by laurent22** [comment]
We need as much info as possible to find the cause for this as it doesn't happen to everybody.

- Please post the content of your Synchronisation Status under the Tools menu (just the top two sections as the folder list is not needed)
- What Markdown plugins are enabled?
- Do you use any custom CSS?

Also if you can think of any detail that might be unusual about your setup, please let me know.

**Comment by jt196** [comment]
@laurent22 as I mentioned in the discourse forum post, I've tried this on a fresh installation (deleted the config folder at least - am I missing some settings somewhere else that may effect this?), and it's still happening after I import my library. 

It doesn't happen before the jex file is imported, and it only happens after the first v1.2. Version 1.1.4 works OK. 

Please post the content of your Synchronisation Status under the Tools menu (just the top two sections as the folder list is not needed) _This does the same with my regular install as well as a fresh one_
What Markdown plugins are enabled? _just the default, I've disabled them all and it's still doing it_
Do you use any custom CSS? _nope_

Would any other files exist from installation that aren't in the config folder and might be causing this? 

Nothing else unusual - as shown by the fresh installation, or what I thought was fresh...

**Comment by jt196** [comment]
Discourse post [here](https://discourse.joplinapp.org/t/delay-lag-while-typing/11789/13)

**Comment by jt196** [comment]
Well did a full clean reinstallation - using CleanMyMac, deleted the config folder and started again. It runs fine with no notes, then I've got slowdown when the import is done. I'll attempt to clean up these tags over the next few days and see if that makes a difference.

**Comment by jt196** [comment]
Cheers for the fix attempt @laurent22 will make sure I grab it as soon as it comes down the pipes and feed back here.

**Comment by jt196** [comment]
![image](https://user-images.githubusercontent.com/8290163/96579448-c6096700-12ce-11eb-9d23-873b18f1edc5.png)

Laurent sorry I didn't understand what you were referring to re the sync status. Here it is. 

FYI I think the sync status is under the Joplin menu these days on at least the 1.3.5 version on Mac.

**Comment by jt196** [comment]
Seems to be solved in release 1.3.7.

**Comment by JackGruber** [comment]
Yes, with v.1.3.7 the problem seems to be gon!
